radixdlt-java/src/main/java/com/radixdlt/client/application/translate/tokens/TokenState.java

```diff
@@ -44,6 +44,28 @@ public TokenState(
 		this.unallocatedTokens = ImmutableMap.copyOf(unallocatedTokens);
 	}
 
+	public static TokenState combine(TokenState state0, TokenState state1) {
+		Map<EUID, UnallocatedTokensParticle> unallocated = new ImmutableMap.Builder<EUID, UnallocatedTokensParticle>()
+				.putAll(state0.unallocatedTokens)
+				.putAll(state1.unallocatedTokens)
+				.build();
+
+		UInt256 unallocatedAmount = unallocated.entrySet().stream()
+			.map(Entry::getValue)
+			.map(UnallocatedTokensParticle::getAmount)
+			.reduce(UInt256.ZERO, UInt256::add);
+
+		return new TokenState(
+			state0.name != null ? state0.name : state1.name,
```

Comment:
So, we state0 takes precedence over state1? Why?

Reply:
it doesn't matter which one because intrinsically they should be the same. This will get cleaned up once we insert the layer in between application state and particles.